src/databricks/labs/ucx/azure/azure_credentials.py

```diff
@@ -0,0 +1,283 @@
+                continue
+            if not azure_sp_info.secret_key:
+                continue
+            secret_value = self._read_databricks_secret(
+                azure_sp_info.secret_scope, azure_sp_info.secret_key, azure_sp_info.application_id
+            )
+            if secret_value:
+                azure_sp_info_with_client_secret[azure_sp_info.application_id] = secret_value
+
+        # update the list of ServicePrincipalMigrationInfo if client_secret is found
+        sp_list_with_secret = []
+        for sp in sp_list:
+            if sp.client_id in azure_sp_info_with_client_secret:
+                sp_list_with_secret.append(
+                    ServicePrincipalMigrationInfo(sp, azure_sp_info_with_client_secret[sp.client_id])
+                )
+        return sp_list_with_secret
+
+    def _print_action_plan(self, sp_list_with_secret: list[ServicePrincipalMigrationInfo]):
+        # print action plan to console for customer to review.
+        for sp in sp_list_with_secret:
+            print(
+                f"Service Principal name: {sp.service_principal.principal}, "
+                f"application_id: {sp.service_principal.client_id}, "
+                f"privilege {sp.service_principal.privilege} "
+                f"on location {sp.service_principal.prefix}"
+            )
+
+    def _generate_migration_list(self):
+        """
+        Create the list of SP that need to be migrated, output an action plan as a csv file for users to confirm
+        """
+        # load sp list from azure_storage_account_info.csv
+        sp_list = self._azure_resource_permissions.load()
+        # list existed storage credentials
+        sc_set = self._list_storage_credentials()
+        # check if the sp is already used in UC storage credential
+        filtered_sp_list = [sp for sp in sp_list if sp.client_id not in sc_set]
+        # fetch sp client_secret if any
+        sp_list_with_secret = self._fetch_client_secret(filtered_sp_list)
+        self._final_sp_list = sp_list_with_secret
+        # output the action plan for customer to confirm
+        self._print_action_plan(sp_list_with_secret)
+
+    def _create_storage_credential(self, sp_migration: ServicePrincipalMigrationInfo):
+        # prepare the storage credential properties
+        name = sp_migration.service_principal.principal
+        azure_service_principal = AzureServicePrincipal(
+            directory_id=sp_migration.service_principal.directory_id,
+            application_id=sp_migration.service_principal.client_id,
+            client_secret=sp_migration.client_secret,
+        )
+        comment = f"Created by UCX during migration to UC using Azure Service Principal: {sp_migration.service_principal.principal}"
+        read_only = False
+        if sp_migration.service_principal.privilege == Privilege.READ_FILES.value:
+            read_only = True
+        # create the storage credential
+        storage_credential = self._ws.storage_credentials.create(
```

Comment:
Do you need to use named arguments everywhere?.

Reply:
fixed